components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/message/OAuthMessage.java

```diff
@@ -0,0 +1,253 @@
+        this.requestType = requestType;
+        if (OAuthAuthorizeState.AUTHENTICATION_RESPONSE.equals(requestType)) {
+            setSessionDataCacheEntry(resultFromLogin);
+        } else if (OAuthAuthorizeState.USER_CONSENT_RESPONSE.equals(requestType)) {
+            setSessionDataCacheEntry(resultFromConsent);
+        }
+    }
+
+    public String getClientId() {
+        return request.getParameter("client_id");
+    }
+
+    public String getSessionDataKeyFromLogin() {
+        return getSessionDataKey(request);
+    }
+
+    public String getSessionDataKeyFromConsent() {
+        return sessionDataKeyFromConsent;
+    }
+
+    public void setSessionDataKeyFromConsent(String sessionDataKeyFromConsent) {
+        this.sessionDataKeyFromConsent = sessionDataKeyFromConsent;
+    }
+
+    public SessionDataCacheEntry getResultFromLogin() {
+        return resultFromLogin;
+    }
+
+    public void setResultFromLogin(SessionDataCacheEntry resultFromLogin) {
+        this.resultFromLogin = resultFromLogin;
+    }
+
+    public SessionDataCacheEntry getResultFromConsent() {
+        return resultFromConsent;
+    }
+
+    public void setResultFromConsent(SessionDataCacheEntry resultFromConsent) {
+        this.resultFromConsent = resultFromConsent;
+    }
+
+    public boolean isForceAuthenticate() {
+        return forceAuthenticate;
+    }
+
+    public void setForceAuthenticate(boolean forceAuthenticate) {
+        this.forceAuthenticate = forceAuthenticate;
+    }
+
+    public boolean isPassiveAuthentication() {
+        return isPassiveAuthentication;
+    }
+
+    public void setPassiveAuthentication(boolean passiveAuthentication) {
+        this.isPassiveAuthentication = passiveAuthentication;
+    }
+
+    public SessionDataCacheEntry getSessionDataCacheEntry() {
+        return sessionDataCacheEntry;
+    }
+
+    public void setSessionDataCacheEntry(SessionDataCacheEntry sessionDataCacheEntry) {
+        this.sessionDataCacheEntry = sessionDataCacheEntry;
+    }
+
+    /**
+     * In federated and multi steps scenario there is a redirection from commonauth to samlsso so have to get
+     * session data key from query parameter
+     *
+     * @param req Http servlet request
+     * @return Session data key
+     */
+    private String getSessionDataKey(HttpServletRequest req) {
+        String sessionDataKey = (String) req.getAttribute(OAuthConstants.SESSION_DATA_KEY);
+        if (sessionDataKey == null) {
+            sessionDataKey = req.getParameter(OAuthConstants.SESSION_DATA_KEY);
+        }
+        return sessionDataKey;
+    }
+
+    public boolean isConsentResponseFromUser() {
+        return resultFromConsent != null;
+    }
+
+    public boolean isAuthResponseFromFramework() {
+        return resultFromLogin != null;
+    }
+
+    public boolean isInitialRequest() {
+        return request.getParameter("client_id") != null && getSessionDataKey(request) == null
```

Comment:
Shall we use a constant for "client_id"?

Reply:
done